app_feup/lib/view/Widgets/schedule_slot.dart

```diff
@@ -92,6 +99,23 @@ class ScheduleSlot extends StatelessWidget {
         TextAlign.center);
   }
 
+  Widget createScheduleSlotClass(context) {
+    final classText = this.classNumber != '' ? (' | ' + this.classNumber) : '';
```

Comment:
I would advise on using a null pattern instead of an empty string, it will probably make for easier conversion whenever null safety finally comes to dart: